lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp

```diff
@@ -1540,12 +1607,398 @@ void amf_app_handle_initial_context_setup_rsp(
+  /* Ciphering algorithms, EEA1 and EEA2 expects length to be mode of 4,
+   * so length is modified such that it will be mode of 4
+   */
+  AMF_GET_BYTE_ALIGNED_LENGTH(len);
+  if (msg.header.security_header_type != SECURITY_HEADER_TYPE_NOT_PROTECTED) {
+    amf_msg_header* header = &msg.security_protected.plain.amf.header;
+    /*
+     * Expand size of protected NAS message
+     */
+    len += NAS_MESSAGE_SECURITY_HEADER_SIZE;
+    /*
+     * Set header of plain NAS message
+     */
+    header->extended_protocol_discriminator = M5GS_MOBILITY_MANAGEMENT_MESSAGE;
+    header->security_header_type = SECURITY_HEADER_TYPE_NOT_PROTECTED;
+  }
+
+  buffer = bfromcstralloc(len, "\0");
+  bytes = nas5g_message_encode(buffer->data, &msg, len,
+                               &ue_context->amf_context._security);
+  if (bytes > 0) {
+    ue_pdu_id_t id = {ue_id, smf_ctx->smf_proc_data.pdu_session_id};
+    buffer->slen = bytes;
+    smf_ctx->session_message = bstrcpy(buffer);
+    pdu_session_resource_modify_request(ue_context, ue_id, smf_ctx, buffer);
+    smf_ctx->retransmission_count = 0;
+    smf_ctx->T3591.id = amf_pdu_start_timer(
+        PDU_SESSION_MODIFICATION_TIMER_MSECS, TIMER_REPEAT_ONCE,
+        pdu_session_resource_modification_t3591_handler, id);
+
+  } else {
+    OAILOG_WARNING(LOG_AMF_APP, "NAS encode failed \n");
+    bdestroy_wrapper(&buffer);
+  }
+
+  return rc;
+}
+void amf_app_fill_create_new_tft(create_new_tft_t* new_tft, QOSRule* qos_rule) {
+  for (int i = 0; i < qos_rule->no_of_pkt_filters; i++) {
+    NewQOSRulePktFilter new_qos_rule_pkt_filter = {};
+    packet_filter_t* pkt_filter = NULL;
+    uint16_t pkt_filter_len = 0;
+    pkt_filter = &new_tft[i];
+    memset(&new_qos_rule_pkt_filter, 0, sizeof(NewQOSRulePktFilter));
+    new_qos_rule_pkt_filter.spare = 0x0;
+    new_qos_rule_pkt_filter.pkt_filter_dir = pkt_filter->direction;
+    new_qos_rule_pkt_filter.pkt_filter_id = pkt_filter->identifier;
+    uint16_t flag = TRAFFIC_FLOW_TEMPLATE_IPV4_REMOTE_ADDR_FLAG;
+    while (flag <= TRAFFIC_FLOW_TEMPLATE_FLOW_LABEL_FLAG) {
+      switch (pkt_filter->packetfiltercontents.flags & flag) {
+        case TRAFFIC_FLOW_TEMPLATE_IPV4_REMOTE_ADDR_FLAG: {
+          /*
+           * IPv4 remote address type
+           */
+          new_qos_rule_pkt_filter.contents[pkt_filter_len] =
+              TRAFFIC_FLOW_TEMPLATE_IPV4_REMOTE_ADDR;
+          pkt_filter_len++;
+
+          for (int j = 0; j < TRAFFIC_FLOW_TEMPLATE_IPV4_ADDR_SIZE; j++) {
+            new_qos_rule_pkt_filter.contents[pkt_filter_len] =
+                pkt_filter->packetfiltercontents.ipv4remoteaddr[j].addr;
+            new_qos_rule_pkt_filter
+                .contents[pkt_filter_len +
+                          TRAFFIC_FLOW_TEMPLATE_IPV4_ADDR_SIZE] =
+                pkt_filter->packetfiltercontents.ipv4remoteaddr[j].mask;
+            pkt_filter_len++;
+          }
+
+          pkt_filter_len += TRAFFIC_FLOW_TEMPLATE_IPV4_ADDR_SIZE;
```

Comment:
Why is pkt_filter_len increase by TRAFFIC_FLOW_TEMPLATE_IPV4_ADDR_SIZE. It is already incremented in line num, 1887 within loop.